RubberduckTests/Inspections/UnreachableCaseInspectionTests.cs

```diff
@@ -0,0 +1,2398 @@
+        [TestCase("10_>_11", "False")]
+        [TestCase("10_>=_10", "True")]
+        [TestCase("5_And_0", "False")]
+        [TestCase("5_Or_0", "True")]
+        [TestCase("5_Xor_6", "False")]
+        [TestCase("6.5_>_5.2", "True")]
+        [Category("Inspections")]
+        public void UciUnit_LogicBinaryConstants(string operands, string expected)
+        {
+            GetBinaryOpValues(operands, out IParseTreeValue LHS, out IParseTreeValue RHS, out string opSymbol);
+
+            var result = Calculator.Evaluate(LHS, RHS, opSymbol);
+
+            Assert.AreEqual(expected, result.ValueText);
+            Assert.IsTrue(result.ParsesToConstantValue);
+        }
+
+        //        Dim A, B, C, D, MyCheck
+        //A = 10: B = 8: C = 6: D = Null    ' Initialize variables.
+        //MyCheck = A > B Eqv B > C    ' Returns True.
+        //MyCheck = B > A Eqv B > C    ' Returns False.
+        //MyCheck = A > B Eqv B > D    ' Returns Null.
+        //MyCheck = A Eqv B    ' Returns -3 (bitwise comparison).
+        [TestCase("True_Eqv_True", "True")]
+        [TestCase("False_Eqv_True", "False")]
+        [TestCase("True_Eqv_False", "False")]
+        [TestCase("False_Eqv_False", "True")]
+        [Category("Inspections")]
+        public void UciUnit_LogicEqvOperator(string operands, string expected)
+        {
+            GetBinaryOpValues(operands, out IParseTreeValue LHS, out IParseTreeValue RHS, out string opSymbol);
+
+            var result = Calculator.Evaluate(LHS, RHS, opSymbol);
+
+            Assert.AreEqual(expected, result.ValueText);
+            Assert.IsTrue(result.ParsesToConstantValue);
+        }
+
+        [TestCase("True_Imp_True", "True")]
+        [TestCase("False_Imp_True", "True")]
+        [TestCase("True_Imp_False", "False")]
+        [TestCase("False_Imp_False", "True")]
+        [Category("Inspections")]
+        public void UciUnit_LogicImpOperator(string operands, string expected)
+        {
+            GetBinaryOpValues(operands, out IParseTreeValue LHS, out IParseTreeValue RHS, out string opSymbol);
+
+            var result = Calculator.Evaluate(LHS, RHS, opSymbol);
+
+            Assert.AreEqual(expected, result.ValueText);
+            Assert.IsTrue(result.ParsesToConstantValue);
+        }
+
+        [TestCase("Not_False", "True")]
+        [TestCase("Not_True", "False")]
+        [Category("Inspections")]
+        public void UciUnit_LogicUnaryConstants(string operands, string expected)
+        {
+            GetUnaryOpValues(operands, out IParseTreeValue theValue, out string opSymbol);
+
+            var result = Calculator.Evaluate(theValue, opSymbol);
+
+            Assert.AreEqual(expected, result.ValueText);
+            Assert.IsTrue(result.ParsesToConstantValue, "Expected IsConstantValue field to be 'True'");
+        }
+
+        [TestCase("-_45", "-45")]
+        [TestCase("-_23.78", "-23.78")]
+        [TestCase("-_True", "True?Boolean")]
+        [TestCase("-_False", "False?Boolean")]
```

Comment:
In both boolean cases, the result should be an `Integer`. In fact, `-True` evaluates to `1` (which is implicitly true if doing a boolean evaluation but will fail for non-boolean equality or comparison).

Reply:
Modified (and added tests) to have the result presented as a requested type (which would generally be the select expression type).